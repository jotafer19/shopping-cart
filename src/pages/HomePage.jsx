import Slideshow from "../components/Slideshow/Slideshow"

export default function HomePage() {
    return (
        <Slideshow />
    )
}