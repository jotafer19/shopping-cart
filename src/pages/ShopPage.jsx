import useFetchData from "../data/fetch"
import CardContainer from "../components/card-container/CardContainer"
import Card from "../components/item-card/Card"
import Loader from "../components/loader/Loader"

export default function ShopPage() {
    const { data, error, isLoading } = useFetchData()
    console.log(data)

    
    if (isLoading) return <Loader />
    if (error) return <p>{error}</p>
    
    return (
        <CardContainer>
            {data.map(item => <Card key={item.id} item={item} />)}
        </CardContainer>
    )
}