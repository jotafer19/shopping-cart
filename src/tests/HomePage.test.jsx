import { render, screen } from "@testing-library/react";
import HomePage from "../pages/HomePage";
import ShopPage from "../pages/ShopPage"
import { MemoryRouter, Routes, Route } from "react-router-dom";
import { beforeEach, expect } from "vitest";
import userEvent from "@testing-library/user-event";

describe("Homepage component", () => {
    beforeEach(() => {
        render(
            <MemoryRouter initialEntries={['/']}>
                <Routes>
                    <Route path="/" element={<HomePage />} />
                    <Route path="/shop" element={<ShopPage />} />
                </Routes>
            </MemoryRouter>
        ) 
    })
    
    it("Renders homepage", () => {
        const { container } = render(<MemoryRouter><HomePage /></MemoryRouter>);

        expect(container).toMatchSnapshot()
    })

    it("Check slideshow is in the document", () => {
        const buttons = screen.getAllByRole("button", {name: "Go to the shop"})
        expect(buttons.length).toBeGreaterThan(0)
        expect(screen.getAllByAltText("Home banner").length).toBeGreaterThan(0)
    })

    it("Go to shop page", async () => {
        const user = userEvent.setup()

        const buttons = screen.getAllByRole("button", {name: "Go to the shop"})

        await user.click(buttons[0])

        setTimeout(() => {
            expect(screen.getByRole("button", { name: "All products" })).toBeInTheDocument()
        }, 1000)
    })
})