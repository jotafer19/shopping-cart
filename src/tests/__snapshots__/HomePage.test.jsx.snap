// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage component > Renders homepage 1`] = `
<div>
  <div
    class="_slideshow_b894e2"
  >
    <div
      class="_slideshow-container_b894e2"
    >
      <div
        class="_slide_b894e2 _active_b894e2"
      >
        <div
          class="_text_b894e2 _rposition_b894e2"
        >
          New collection
          <a
            class="_link_b894e2"
            href="/shop"
          >
            <button>
              Go to the shop
            </button>
          </a>
        </div>
        <img
          alt="Home banner"
          class="_image_b894e2"
          src="/src/assets/images/shop-image-1.jpg"
        />
      </div>
      <div
        class="_slide_b894e2 "
      >
        <div
          class="_text_b894e2 _lposition_b894e2"
        >
          New products to enjoy this summer
          <a
            class="_link_b894e2"
            href="/shop"
          >
            <button>
              Go to the shop
            </button>
          </a>
        </div>
        <img
          alt="Home banner"
          class="_image_b894e2"
          src="/src/assets/images/shop-image-2.jpg"
        />
      </div>
      <div
        class="_slide_b894e2 "
      >
        <div
          class="_text_b894e2 _rposition_b894e2"
        >
          Update your devices!
          <a
            class="_link_b894e2"
            href="/shop"
          >
            <button>
              Go to the shop
            </button>
          </a>
        </div>
        <img
          alt="Home banner"
          class="_image_b894e2"
          src="/src/assets/images/shop-image-3.jpg"
        />
      </div>
      <div
        class="_slide_b894e2 "
      >
        <div
          class="_text_b894e2 _lposition_b894e2"
        >
          The best gifs for your friends
          <a
            class="_link_b894e2"
            href="/shop"
          >
            <button>
              Go to the shop
            </button>
          </a>
        </div>
        <img
          alt="Home banner"
          class="_image_b894e2"
          src="/src/assets/images/shop-image-4.jpg"
        />
      </div>
    </div>
    <div
      class="_buttons-container_b894e2"
    >
      <button
        class="_left_b894e2"
      >
        <img
          alt="Previous image"
          src="/src/assets/icons/left-arrow.svg"
        />
      </button>
      <button
        class="_right_b894e2"
      >
        <img
          alt="Next image"
          src="/src/assets/icons/right-arrow.svg"
        />
      </button>
    </div>
  </div>
</div>
`;
